node_modules/credscan-pkg/main.js

```diff
@@ -0,0 +1,52 @@
+const path = require('path');
+const config = require('./config.json');
+var edge = require('edge-js');
+var core = require('@actions/core');
+
+var version = 'coreapp'
+var namespace = 'CredScanSrc'
+
+const baseNetAppPath = path.join(__dirname, '/src/' + namespace + '/bin/Debug/net' + version + '2.0');
+
+process.env.EDGE_USE_CORECLR = 1;
+
+var baseDll = path.join(baseNetAppPath, namespace + '.dll');
+var localTypeName = namespace + '.LocalMethods';
+
+var credScanFunction = edge.func({
+    assemblyFile: baseDll,
+    typeName: localTypeName,
+    methodName: 'Invoke'
+});
+
+async function credscan(input, scannedResult) {
+    if (!config.credScanEnable) return;
+    if (process.env.RUNNER_OS == 'macOS') {
```

Comment:
we should also ensure that dotnet runtime is there while running.

Reply:
I looked for this online but have not been able to find anything to do from Node.js. Should I add this check to the preinstall file?

Reply:
Host for building and running can be different.

Reply:
I'll remove the check